components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/APIStateChangeWorkflow.java

```diff
@@ -63,13 +65,15 @@
     private ApiDAO apiDAO;
     private APISubscriptionDAO apiSubscriptionDAO;
     private APILifecycleManager apiLifecycleManager;
+    private APIGateway apiGateway;
 
     public APIStateChangeWorkflow(ApiDAO apiDAO, APISubscriptionDAO apiSubscriptionDAO, WorkflowDAO workflowDAO,
-                                  APILifecycleManager apiLifecycleManager) {
+                                  APILifecycleManager apiLifecycleManager, APIGateway apiGateway) {
```

Comment:
@dewmini : But components need to follow Version strategy. So the version for component changing in the patch range is wrong.